pkg/apis/core/validation/validation_test.go

```diff
@@ -10112,6 +10112,28 @@ func TestValidateService(t *testing.T) {
 			},
 			numErrs: 1,
 		},
+		{
+			name: "valid, service with no IPFamily",
```

Comment:
should not be valid if we have defaulting

Reply:
unless we want to do away with "Default" and just document "unspecified" as "do the right thing", but we have precedent elsewhere for "Default".  I'll ask API reviewers for an opinion.  Your thoughts?

Reply:
api-reviewers agree NOT to rely on `nil` meaning "Default".  We have defaulting logic, let's rely on it.

Reply:
The main purpose of "Default" or "ClusterDefault" is to allow for clusters to be upgraded. `unspecified` means a different  thing from `default`. It is implicitly specified to be default :-) 

As for having having a test (even with defaulting): it is `test until fear turns to boredom` nothing more. I can always remove it, if needed.

Reply:
It's obnoxious to have to set a bunch of "optional-to-users-but-required-for-validation" fields in tests, but really we should not be explicitly allowing invalid values.  This is why I like the "tweak function" table tests.

Reply:
I removed the test